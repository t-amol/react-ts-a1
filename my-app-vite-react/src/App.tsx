import React, { Suspense } from "react";

import ListOfCourses from "./listofcourses/listofcourses";
// import Posts from "./posts/posts";
import GetPostById from "./getpostbyid/getpostbyid";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import Navbar from "./navbar/navbar";
import CourseDetails from "./coursedetails/coursedetails";
import NewCourse from "./newcourse/newcourse";
import GrandParent from "./contextapi/contextapi";
import { SignUpBootstrap4 } from "./forms/uniforms/signupbootstrap4";
import { SignUpAntd } from "./forms/uniforms/signupantd";
import { SignUpBootstrap5 } from "./forms/uniforms/signupbootstrap5";
import { SignUpMui } from "./forms/uniforms/signupmui";
import { SignUpSemantic } from "./forms/uniforms/signupsemantic";
import { SignUpUnstyled } from "./forms/uniforms/signupunstyled";
import { FormJsonform } from "./forms/jsonforms/formjsonform";
import { TableAgGrid } from "./tables/tableaggrid/tableaggrid";

const Posts = React.lazy(() => import("./posts/posts"));

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <Navbar />
      <Routes>
        <Route path="/" Component={ListOfCourses}></Route>
        <Route path="/newcourse" Component={NewCourse}></Route>
        
{/*         <Route path="/formUnstyled" Component={FormUnstyled}></Route>
        <Route path="/signupantd" Component={SignUpAntd}></Route>
        <Route path="/signupbootstrap4" Component={SignUpBootstrap4}></Route>
        <Route path="/signupbootstrap5" Component={SignUpBootstrap5}></Route> */}
        <Route path="/signupmui" Component={SignUpMui}></Route>
{/*         <Route path="/signupsemantic" Component={SignUpSemantic}></Route>
        <Route path="/signupunstyled" Component={SignUpUnstyled}></Route> */}

        <Route path="/formjsonform" Component={FormJsonform}></Route>
        <Route path="/tableaggrid" Component={TableAgGrid}></Route>
        
        <Route path="/coursedetails/:cid" Component={CourseDetails}></Route>
        <Route path="/contextapi" Component={GrandParent}></Route>
        <Route
          path="/posts"
          element={
            <Suspense fallback={<p>Loading...</p>}>
              <Posts />
            </Suspense>
          }
        ></Route>
        <Route path="/getpostbyid" Component={GetPostById}></Route>

        <Route
          path="*"
          element={
            <>
              <h1>Page Not Found !</h1>
              <img src="https://internetdevels.com/sites/default/files/public/blog_preview/404_page_cover.jpg" />
            </>
          }
        ></Route>
      </Routes>
    </BrowserRouter>
  );
};

export default App;

// class App extends React.Component {
//   render() {
//     return <ListOfCourses />;
//   }
// }
// export default App;
