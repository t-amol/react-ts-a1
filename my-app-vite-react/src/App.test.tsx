import { render, screen} from '@testing-library/react';
import renderer from "react-test-renderer";
import App from './App';

describe('App Component', () => {
  test("renders learn react link", () => {
    render(<App />);
    const linkElement = screen.getByText(/learn react/i);
    expect(linkElement).toBeInTheDocument();
  });

  test("test snapshot", () => {
    render(<App />);
    const linkElement = screen.getByTitle(/apptitle/i);
    //const domTree = renderer.create(<App />).toJSON();
    expect(linkElement).toMatchSnapshot();
  });

});
