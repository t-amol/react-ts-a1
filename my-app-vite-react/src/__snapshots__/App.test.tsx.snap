// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component test snapshot 1`] = `
<div
  class="app"
  title="apptitle"
>
  <div>
    <a
      href="https://vite.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="[object Object]"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="[object Object]"
      />
    </a>
  </div>
  <h1>
    hello_world
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      0
    </button>
    <button>
      English
    </button>
    <button>
      Fran√ßais
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
    <a
      class="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
